### PR TITLE
47132 vaos va link/confirmed appointment detail page

### PR DESCRIPTION
**Pull request by ryanshaw**
## Description
This is one of several PRs that replace the react-router `Link` component in VAOS with the newer `va-link` web component.

[va-link](https://design.va.gov/components/link/)
[va-link storybook](https://design.va.gov/storybook/?path=/docs/components-va-link--default)

This was broken up into multiple PR's to allow for easier code review as this change impacts nearly every file and every test within VAOS.

## Original issue(s)
[department-of-veterans-affairs/va.gov-team/issues/47132](https://github.com/department-of-veterans-affairs/va.gov-team/issues/47132)

## Related to these other PR's:
- [requested appointments](https://github.com/department-of-veterans-affairs/vets-website/pull/22776)
- [breadcrumbs](https://github.com/department-of-veterans-affairs/vets-website/pull/22779)
- [covid flow](https://github.com/department-of-veterans-affairs/vets-website/pull/22818)
- [new appointment](https://github.com/department-of-veterans-affairs/vets-website/pull/23125)
- [appointment list](https://github.com/department-of-veterans-affairs/vets-website/pull/23128)

## Testing done
- Unit testing
- e2e testing

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/StatusAlert.jsx [hunk 1]
```diff
@@ -54,12 +54,19 @@ export default function StatusAlert({ appointment, facility }) {
         <strong>We’ve scheduled and confirmed your appointment.</strong>
         <br />
         <div className="vads-u-margin-y--1">
-          <Link to="/" onClick={handleClick()}>
-            Review your appointments
-          </Link>
+          <va-link
+            href="/"
```
Comment (jenniemc) [inline src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/StatusAlert.jsx:58]: This link is going to the wrong path. I think "/" was enrolled in a special routes file. @vbahinwillit may have the background on it.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/StatusAlert.jsx [hunk 2]
```diff
@@ -54,12 +54,19 @@ export default function StatusAlert({ appointment, facility }) {
...
+            onClick={handleClick()}
+            text="Review your appointments"
+            data-testid="review-appointments-link"
+          />
         </div>
         <div>
-          <Link to="/new-appointment">Schedule a new appointment</Link>
+          <va-link
+            href="/new-appointment"
```
Comment (jenniemc) [inline src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/StatusAlert.jsx:66]: This link is going to the wrong path. "/new-appointment" was enrolled in a special routes file. I don't know how it works but @vbahinwillit may have the background on it.

**Review by jenniemc — CHANGES_REQUESTED** [review]
Looks like the links on the status alert page are also experiencing the same wrong path.

**Review by vbahinwillit — APPROVED** [review]
LGTM!

**Comment by ryanshaw** [comment]
> Looks like the links on the status alert page are also experiencing the same wrong path.

@jenniemc I changed this to use the va-link component with the React bindings to handle the onclick event.

**Pull request closed without merging**